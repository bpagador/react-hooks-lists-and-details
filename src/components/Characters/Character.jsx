import React from 'react';
import PropTypes from 'prop-types';

const Character = ({ name, image }) => {
  return (
    <section>
      <h4>{name}</h4>
      <img src={image} alt={name} width="250" />
    </section>
  );
};

Character.propTypes = {
  name: PropTypes.string.isRequired,
  image: PropTypes.string.isRequired
};
