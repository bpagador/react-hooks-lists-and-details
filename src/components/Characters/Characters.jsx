import React from 'react';
import PropTypes from 'prop-types';
import { Link } from 'react-router-dom';
import Character from './Character';

const Characters = ({ characters }) => {
  const characterList = characters.map(character => (
    <li key={character.name}>
      <Link to={`/${character.name}`}>
        <Character {...character} />
      </Link>
    </li>
  ));

  return (
    <ul>
      {characterList}
    </ul>
  );
};

Characters.propTypes = {
  characters: PropTypes.arrayOf(PropTypes.shape({
    name: PropTypes.string.isRequired,
    image: PropTypes.string.isRequired
  })).isRequired 
};

export default Characters;
