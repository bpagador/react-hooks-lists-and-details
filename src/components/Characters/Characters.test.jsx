import React from 'react';
import { shallow } from 'enzyme';
import Characters from './Characters';


describe('many characters', () => {
  it('returns many characters', () => {
    const characters = [{
      name:'Shelley', 
      image:'https://vignette.wikia.nocookie.net/heyarnold/images/c/cb/Shelley.jpg/revision/latest/scale-to-width-down/310?cb=20120527025256'
    }, {
      name:'Mitzi',
      image:'https://vignette.wikia.nocookie.net/heyarnold/images/d/d9/Mitzi.png/revision/latest/scale-to-width-down/310?cb=20171212070331' }
    ];
    const wrapper = shallow(
      <Characters characters={characters}/>
    );
    expect(wrapper).toMatchSnapshot();
  });
});
