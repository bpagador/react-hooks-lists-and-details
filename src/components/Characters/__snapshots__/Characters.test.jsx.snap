// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`many characters returns many characters 1`] = `
<ul>
  <li
    key="Shelley"
  >
    <Link
      to="/Shelley"
    >
      <Character
        image="https://vignette.wikia.nocookie.net/heyarnold/images/c/cb/Shelley.jpg/revision/latest/scale-to-width-down/310?cb=20120527025256"
        name="Shelley"
      />
    </Link>
  </li>
  <li
    key="Mitzi"
  >
    <Link
      to="/Mitzi"
    >
      <Character
        image="https://vignette.wikia.nocookie.net/heyarnold/images/d/d9/Mitzi.png/revision/latest/scale-to-width-down/310?cb=20171212070331"
        name="Mitzi"
      />
    </Link>
  </li>
</ul>
`;
